main/lsp/NextMethodFinder.cc

```diff
@@ -52,20 +52,12 @@ ast::ExpressionPtr NextMethodFinder::preTransformMethodDef(core::Context ctx, as
 
     auto currentMethod = methodDef.symbol;
 
-    auto &currentMethodLocs = currentMethod.data(ctx)->locs();
-    auto inFileOfQuery = [&](const auto &loc) { return loc.file() == this->queryLoc.file(); };
-    auto maybeCurrentLoc = absl::c_find_if(currentMethodLocs, inFileOfQuery);
-    if (maybeCurrentLoc == currentMethodLocs.end()) {
-        return tree;
-    }
-
-    auto currentLoc = *maybeCurrentLoc;
+    auto currentLoc = ctx.locAt(tree.loc());
     if (!currentLoc.exists()) {
         // Defensive in case location information is disabled (e.g., certain fuzzer modes)
         return tree;
     }
 
-    ENFORCE(currentLoc.file() == this->queryLoc.file());
```

Comment:
How sad are we about losing this?

Reply:
Before, it was non obvious that the Loc would be for the right file because we were searching for locs in the symbol table.

now, it’s correct by construction, we only ever operate on LocOffset, and there is only one file in play, so I figured these enforces were one step removed from asserting something like  1 == 1